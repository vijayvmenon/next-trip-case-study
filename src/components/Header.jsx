import React from 'react';
import { createUseStyles } from 'react-jss';
import constants, { colors } from '../constants';

const useStyles = createUseStyles({
  headerRoot: {
    background: colors.BACKGROUND_2,
    height: 40,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    padding: [[8, 0, 8, 24]],
  },
  headerText: {
    color: colors.WHITE,
  },
});
export default function Header() {
  const classes = useStyles();
  return (
    <header className={classes.headerRoot}>
      <h1 className={classes.headerText}>{constants.HEADER_TEXT}</h1>
    </header>
  );
}
