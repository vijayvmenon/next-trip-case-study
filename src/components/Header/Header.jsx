/* eslint-disable */
import React from 'react';
import { createUseStyles } from 'react-jss';
import { useLocation } from 'wouter';
import constants, { colors } from '../../constants';
import { ReactComponent as AppIcon } from '../../assets/app-logo.svg';
import SelectLanguage from '../LanguageSelector/SelectLanguage';

const useStyles = createUseStyles({
  headerRoot: {
    background: colors.BACKGROUND_2,
    height: 40,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    padding: [[16, 0, 16, 12]],
  },
  headerText: {
    color: colors.WHITE,
    '@media(max-width:460px)': {
      fontSize: '1.2rem',
    },
  },
  appLogo: {
    width: 40,
    height: 40,
    paddingRight: 8,
    cursor: 'pointer',
    '@media(max-width:460px)': {
      width: 30,
      height: 30,
    },
  },
});
export default function Header() {
  const classes = useStyles();
  const [, setLocation] = useLocation();

  return (
    <header className={classes.headerRoot}>
      <AppIcon onClick={() => setLocation('/')} className={classes.appLogo} />
      <h1 className={classes.headerText}>{constants.HEADER_TEXT}</h1>
      <SelectLanguage />
    </header>
  );
}
