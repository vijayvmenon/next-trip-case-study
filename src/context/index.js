import { useContext } from 'react';
import AppContext, { Context } from './context';

export default AppContext;
export { Context };
export const useMyContext = () => useContext(Context);
