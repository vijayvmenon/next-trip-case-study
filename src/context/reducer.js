import constants from '../constants';
import ENTranslations from '../locales/en.json';
import { localStorageExists } from '../utils';

const localStorageLanguage =
  localStorageExists && localStorage.getItem('LANGUAGE');
const initialState = {
  tableExpanded: false,
  hideHeader: false,
  snackbar: { value: '', text: '' },
  language: localStorageLanguage || 'en',
  translations: localStorageLanguage
    ? constants.LANGUAGE_MAPPING[localStorageLanguage]
    : ENTranslations,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case 'TABLE_EXPAND':
      return { ...state, tableExpanded: true };
    case 'TABLE_COLLAPSE':
      return { ...state, tableExpanded: false };
    case 'SET_SNACKBAR':
      return { ...state, snackbar: { value: action.value, text: action.text } };
    case 'HIDE_HEADER':
      return { ...state, hideHeader: true };
    case 'SHOW_HEADER':
      return { ...state, hideHeader: false };
    case 'SET_LANGUAGE':
      return {
        ...state,
        language: action.value,
        translations: constants.LANGUAGE_MAPPING[action.value],
      };
    default:
      return { ...state };
  }
}

export { initialState };
