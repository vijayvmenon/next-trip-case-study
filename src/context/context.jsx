import React, { useReducer, createContext } from 'react';
import PropTypes from 'prop-types';
import reducer, { initialState } from './reducer';

export const Context = createContext();

export default function AppContext({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
}

AppContext.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,
};
