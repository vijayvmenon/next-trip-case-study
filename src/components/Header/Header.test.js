import Header from './Header';
import renderer from 'react-test-renderer';
import { Context } from '../../context';

const dispatch = jest.fn(); // mock dispatch

describe('Component:Header.jsx', () => {
  test('Header renders correctly', () => {
    const headerTree = renderer
      .create(
        // Pass Context as SelectLanguage uses Context
        <Context.Provider value={{ dispatch }}>
          <Header />,
        </Context.Provider>,
      )
      .toJSON();
    expect(headerTree).toMatchSnapshot();
  });
});
