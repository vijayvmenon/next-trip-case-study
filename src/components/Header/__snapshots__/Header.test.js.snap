// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Header.jsx Header renders correctly 1`] = `
Array [
  <header
    className="headerRoot-0-2-1"
  >
    <svg
      className="appLogo-0-2-3"
      onClick={[Function]}
    >
      app-logo.svg
    </svg>
    <h1
      className="headerText-0-2-2"
    >
      Minneapolis Metro Transit
    </h1>
    <div
      className="root-0-2-4"
    >
      <div
        className="rootLanguage-0-2-6"
      >
        <select
          className="languageSelect-0-2-13 languageSelected-0-2-8"
          data-testid="custom-select-id"
          onChange={[Function]}
          value="en"
        >
          <option
            data-testid="option-default-id"
            disabled={true}
            hidden={true}
            value="Select Language"
          />
          <option
            className="selectedOption-0-2-15"
            data-testid="option-en-id"
            value="en"
          >
            English
          </option>
          <option
            data-testid="option-es-id"
            value="es"
          >
            Spanish
          </option>
        </select>
      </div>
    </div>
  </header>,
  ",",
]
`;
